glean-core/android/src/test/java/mozilla/telemetry/glean/GleanTest.kt

```diff
@@ -883,4 +886,46 @@ class GleanTest {
         assertEquals("foo", GleanInternalMetrics.appDisplayVersion.testGetValue())
         assertEquals("2020-11-06T11:30:50+00:00", GleanInternalMetrics.buildDate.testGetValueAsString())
     }
+
+    @Test
+    fun `Date header is set on actualy HTTP POST`() {
+        delayMetricsPing(context)
+        val server = getMockWebServer()
+        resetGlean(
+            context,
+            Glean.configuration.copy(
+                serverEndpoint = "http://" + server.hostName + ":" + server.port
+            )
+        )
+
+        Glean.handleBackgroundEvent()
+
+        // Before the `Date` was set on submit,
+        // so waiting here would ensure the actual sending is later.
+        // Now we set the date on actual sending,
+        // so we can observe it close to receive time further below.
+        Thread.sleep(2000)
```

Comment:
Not necessarily a fan of intentionally slowing down our tests, but because this is all about time measurement we don't really have another better way